app/javascript/orangelight/google_books_snippets.es6

```diff
@@ -0,0 +1,42 @@
+import { insert_online_link } from 'orangelight/insert_online_link'
+
+export default class GoogleBooksSnippets {
+  get has_figgy_viewer() {
+      return $(".document-viewers > .intrinsic-container").length > 0
+  }
+
+  get isbn() {
+    return $("meta[property='isbn']").map(function(){ return $(this).attr("content")}).toArray()
+  }
+
+  get googleUrl() {
+    return `https://books.google.com/books?callback=?&jscmd=viewapi&bibkeys=://books.google.com/books?callback=?&jscmd=viewapi&bibkeys=${this.isbn.join(",")}`
+  }
+
+  async insert_snippet() {
+    return $.getJSON(this.googleUrl).promise()
+      .then(this.process_google_response.bind(this))
+  }
+
+  get contentStatement() {
+    return "Google Books contains a limited preview of this book, which may be in copyright and may not be available in electronic form through other services offered by Princeton University Library (PUL). We are sharing this link with an intent to make access to information as efficient as possible while physical access to our collections is limited. This item may still be requested for pickup or electronic fulfillment via the library catalog. PUL’S TEMPORARY USE OF THIS SERVICE SHOULD NOT BE INTERPRETED AS A RECOMMENDATION OR ENDORSEMENT OF ANY PRODUCTS OR SERVICES, INCLUDING ANY PRODUCTS OR SERVICES PRESENTED IN THIRD-PARTY ADVERTISEMENTS ON THIS SITE."
+
+  }
+
+  process_google_response(response) {
+    for(const key in response) {
+      const result = response[key]
+      if(result.preview === "partial" || result.preview === "full")
+      {
+        const url = new URL(result.preview_url)
+        const bookId = url.searchParams.get("id")
+        const link = `https://www.google.com/books/edition/_/${bookId}?hl=en&gbpv=1&pg=PP1`
+        const content = (link, target) => {
+          return `<a href="${link}" target="${target}">Google Preview</a><p>${this.contentStatement}</p>`
```

Comment:
Specify "partial" or "full" in the link text?

Reply:
If it's full is it even a preview anymore? Possible names fluttering in my head are `Google Books (Partial)`, `Google Preview (Partial)`, `Google Book Viewer (Partial)`, `Google Book Viewer`. Maybe input from @escowles or @jpstroop ?

Reply:
I thought we we scoping to previews only and that the full views would be covered by Hathi, but you're right, there are probably exceptions. `Google Books (Partial View)` and `Google Books (Full View)` makes sense to me. I'll look at the disclaimer text too and see if that needs adjusting.

Reply:
Disclaimer updated once again: https://docs.google.com/document/d/1xbjEzo4p0vcblY9-KE6Fsv73kNHj151iJysYMCSJnvc/edit

Reply:
![Screen Shot 2020-09-16 at 10 08 22 AM](https://user-images.githubusercontent.com/2806645/93369692-9b764b00-f804-11ea-8f72-8fa6070ac37f.png)